test/annexB/language/statements/for-in/nonstrict-initializer.js

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-initializers-in-forin-statement-heads
+description: >
+    for-in initializers in nonstrict mode
+flags: [noStrict]
+---*/
+(function() {
+  var effects = 0;
+  for (var a = ++effects in {});
+  assert.sameValue(effects, 1);
+})();
+
+
+(function() {
+  var stored;
+  for (var a = 0 in stored = a, {});
+  assert.sameValue(stored, 0);
+})();
+
+
+(function() {
+  for (var a = 0 in {});
+  assert.sameValue(a, 0);
+})();
+
+
+(function() {
+  var effects = 0;
+  var iterations = 0;
+  var stored;
+  for (var a = (++effects, -1) in stored = a, {a: 0, b: 1, c: 2}) {
+    ++iterations;
+  }
+  assert.sameValue(stored, -1);
+  assert.sameValue(effects, 1);
+  assert.sameValue(iterations, 3);
```

Comment:
I like these tests, but I believe we can improve the assertions providing a message to identify each one. This would be helpful if any fails.